### ISSUE TITLE
Add option to ignore labels in `skaffold render`

### ISSUE THREAD

**Issue by nkubala** [GoogleContainerTools/skaffold#4228]
we should add the ability to exclude runner-specific labels from the rendered manifests. some of these are nice to have as metadata on pods created during development, but aren't very useful for rendered manifests that get committed to git repos. we can still leave the `app.kubernetes.io/managed-by` label.

`skaffold render --no-labels`

**Comment by alewis001** [comment]
Seeing as my question in slack prompted this, the least I can do is have a go at contributing it :) 

I've not done any development in Go before but I'll do my best to get this done soon; it helps that it's marked as "good first issue", that gives me some confidence ;)

**Comment by alewis001** [comment]
@nkubala I have a branch where I've made some progress (hopefully you'll see it as progress): [alewis001/skaffold/feature-4105](https://github.com/alewis001/skaffold/tree/feature-4105)

This change filters the skaffold.dev labels for the kustomize deployer.

I tried to keep the changes as isolated as possible to avoid too many things having to know that the what the `no-labels` flag is, and whether or not it was set. As each of the deployers seemed to have the similar logic in their `renderManifests` functions, I thought that seemed like a reasonable place to put the label filtering.

What's the best way to get feedback? Open a PR? 

If I'm in the right direction then I need to:

- Add tests
- Add the label filtering to the other deployers.

I'd like to get some feedback that I'm going in the right direction before I go too far down the wrong road.
